src/neuroconv/tools/spikeinterface/spikeinterface.py

```diff
@@ -633,7 +634,9 @@ def add_electrical_series(
         else:
             eseries_kwargs.update(conversion=1e-6)
             eseries_kwargs.update(channel_conversion=channel_conversion)
-    if iterator_type is None or iterator_type == "v2":
+    if iterator_type is None:
+        ephys_data = checked_recording.get_traces(return_scaled=write_scaled, segment_index=segment_index)
```

Comment:
I would strongly recommend adding perhaps a `psutil` check of available memory (similar to the movie interface) to ensure the user does not load in significantly more data than they really ought to be capable of

Reply:
Good idea, I will check it out.

Reply:
Done. I added a test for this functionality as well using mocks.